### PR TITLE
Hardcode LC_CTYPE for lowerstr_with_len

### PR DESCRIPTION
**Pull request by mwiencek**
This is what the production database's Ctype was previously set to. It affects the dictionary normalization, because `lowerstr_with_len` doesn't handle wide chars when the "C" locale is used (i.e. what the production database is currently using).

So, the current behavior is that `musicbrainz_dunaccentdict_lexize` doesn't lowercase any wide chars. While it would help in some cases to reverse the order in which `lowerstr_with_len` and `unaccent_string` are called (e.g. "Æ" could unaccent to AE, which `lowerstr_with_len` could then handle even in the "C" locale), that doesn't work for wide chars which can be lowercased but not unaccented (e.g. "П"). That change is not as simple as reversing the calls, anyway.

Hardcoding the locale isn't exactly ideal, but it restores the previous dictionary normalization behavior, and we can't know the actual context (locale) of the strings being passed in anyway.

### CODE REVIEW

**Comment by mwiencek** [comment]
@uklauer maybe you have an opinion on this (not sure who else is going to review it :)). It fixes the MBS pgtap test failures in unaccent.sql.

**Comment by mwiencek** [comment]
This'll require dropping and recreating all of our mb_simple search indexes AFAICT.

**Comment by uklauer** [comment]
I’m a bit wary regarding the context in which this runs. Will the locale change be limited to the extension, or will it affect other parts of the PostgreSQL server, too? (Perhaps differently depending on whether a query involving `musicbrainz_unaccent` was run in this process or not?)

If it can be limited to the extension, I’m fine with the change; though I think, in the longer run, we should try to get rid of these custom extensions altogether. I elaborated a bit in a comment on MBS-9084. (For `musicbrainz_collate`, it may even be the case that it can be replaced by built-in `COLLATE` now.)

**Comment by y-van-z** [comment]
As is, it affects subsequent SQL requests as well, for example `SELECT lower ('Ænima');` returns `ænima`, but it does not affect other processes ran on the same database. A way to limit this change strictly to the extension would be to wrap the only `lowerstr_with_len` call with locale set and restore.

**Comment by y-van-z** [comment]
What is the reason for not setting `LC_CTYPE` to `en_US.UTF-8` for the `musicbrainz_db` (except that it requires to dump/reload the entire database)? I could not figure it from doc, tickets and commits log.

**Comment by mwiencek** [comment]
Thanks to both of you for the reviews, I'll make sure the change is confined to the `lowerstr_with_len` call only.

@y-van-z The default setting is `C` (see https://github.com/metabrainz/musicbrainz-server/blob/20bad692b76702d626ef45c878bedd579148b18d/admin/InitDb.pl#L221). It was previously set to `en_US.UTF-8` in production by someone manually, without explanation, though I can perhaps understand this as one of the reasons now. I still think `C` is probably a correct default, since the database isn't specific to that one locale.

File: musicbrainz_unaccent.c [hunk 1]
```diff
@@ -306,7 +306,24 @@ musicbrainz_dunaccentdict_lexize(PG_FUNCTION_ARGS)
...
+       created. "en_US.UTF-8" was historically the Ctype used on the
+       MusicBrainz production database, but that was later switched to "C",
+       causing `lowerstr_with_len` to not handle wide characters.
+
+       The locale is only changed temporarily, as shown in the example here:
+       https://www.gnu.org/software/libc/manual/html_node/Setting-the-Locale.html
+     */
+    char *old_locale = strdup(setlocale(LC_CTYPE, NULL));
+    if (!old_locale) {
+        PG_RETURN_POINTER(NULL);
```
Comment (mwiencek) [inline musicbrainz_unaccent.c]: Tested that and it returns NULL as expected (doesn't crash the server or anything).

**Pull request merged**